linalg/vector.hpp

```diff
@@ -393,6 +393,9 @@ class Vector
    { return mfem::ReadWrite(data, size, false); }
 
 #ifdef MFEM_USE_SUNDIALS
+   // TODO: I think this should be removed, but not sure if it might break old code.
```

Comment:
Highlighting this code comment for discussion. With the addition of the `SundialsNVector` class, I don't think this is needed, at least for the interface to SUNDIALS.

Reply:
I agree but don't think it is necessary to do in this PR. It would be good enough to have in a shortly-after follow up.

Reply:
Unless this constructor is now broken, we should keep it for now and just mark it as deprecated. For an example how we do that see https://github.com/mfem/mfem/blob/4f521a1be6624826ff25924977b43e5667b9ac1c/fem/coefficient.hpp#L149-L152